backend/src/main/java/com/google/sps/notifs/Notifier.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main.java.com.google.sps.notifs;
+
+import com.google.api.services.gmail.Gmail;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+public abstract class Notifier {
```

Comment:
What other implementations did you have in mind for Notifier besides the gmail notifier? I'm still unsure if this should be an abstract class or if all the functionality should be merged into the gmail notifier.

Reply:
+1 I think we can just use the Gmail API for emails (no texts, etc.)

Reply:
I was thinking of adding text messages and potentially more but given the scope of the api's that we are limited to I think this is a good suggestion